lib/puppet/provider/package/apt.rb

```diff
@@ -130,4 +164,27 @@ def purge
   def install_options
     join_options(@resource[:install_options])
   end
+
+  def insync?(is)
```

Comment:
I'm not 100% sure but is it possible we need this line here as well: https://github.com/puppetlabs/puppet/pull/7994/files#diff-41bbce95c82bdc373da03bb2f6116ef5R281 ?